debezium-server-instructlab/src/main/java/io/debezium/server/instructlab/InstructLabSinkConsumer.java

```diff
@@ -0,0 +1,113 @@
+        implements ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InstructLabSinkConsumer.class);
+
+    @Inject
+    DebeziumServer server;
+
+    @PostConstruct
+    void configure() {
+    }
+
+    @PreDestroy
+    void close() {
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        // Read each event and build file batch data
+        final Map<String, QnaFile> batchFiles = new HashMap<>();
+        for (ChangeEvent<Object, Object> record : records) {
+            if (record.value() != null) {
+                // LOGGER.info("Received event {} = '{}'", getString(record.key()), getString(record.value()));
+                getQnaMap(record).forEach((key, attributes) -> {
+                    final String fileName = attributes.get(AbstractAddQnaHeader.ATTRIBUTE_FILENAME);
+                    if (!Strings.isNullOrEmpty(fileName)) {
+                        final String question = attributes.get(AbstractAddQnaHeader.ATTRIBUTE_QUESTION);
+                        final String answer = attributes.get(AbstractAddQnaHeader.ATTRIBUTE_ANSWER);
+                        if (!Strings.isNullOrEmpty(question) && !Strings.isNullOrEmpty(answer)) {
+                            final QnaFile file = batchFiles.computeIfAbsent(fileName, QnaFile::new);
+                            file.addSeedExample(question, answer, attributes.get(AbstractAddQnaHeader.ATTRIBUTE_CONTEXT));
+                        }
+                    }
+                });
+            }
+            committer.markProcessed(record);
+        }
+
+        // Flush files
+        for (QnaFile file : batchFiles.values()) {
+            try {
+                file.flush();
+            }
+            catch (IOException e) {
+                throw new DebeziumException("Failed to flush file: " + file.getFileName(), e);
+            }
+        }
+
+        // Mark finished
+        committer.markBatchFinished();
+    }
+
+    private Map<String, Map<String, String>> getQnaMap(ChangeEvent<Object, Object> record) {
+        final Map<String, Map<String, String>> headers = new HashMap<>();
+        EmbeddedEngineChangeEvent<Object, Object, Object> event = (EmbeddedEngineChangeEvent<Object, Object, Object>) record;
+        Headers kafkaHeaders = event.sourceRecord().headers();
+        for (Header header : kafkaHeaders) {
+            final String key = header.key();
+            if (key.startsWith(AbstractAddQnaHeader.QNA_PREFIX)) {
+                String[] keyParts = key.split("\\.", 3);
+                if (keyParts.length == 3) {
+                    final String prefix = keyParts[1];
+                    final String attribute = keyParts[2];
+                    headers.computeIfAbsent(prefix, k -> new HashMap<>()).put(attribute, String.valueOf(header.value()));
+                }
```

Comment:
Maybe add `else` branch with printing debug message that key has wrong format and record will be skipped?

Reply:
Added a new log entry as requested, LMK if that fits the bill.

Reply:
yes, it does